pulp_ansible/app/viewsets.py

```diff
@@ -267,6 +481,112 @@ class AnsibleRepositoryViewSet(RepositoryViewSet, ModifyRepositoryActionMixin):
     queryset = AnsibleRepository.objects.all()
     serializer_class = AnsibleRepositorySerializer
 
+    queryset_filtering_required_permission = "ansible.view_ansiblerepository"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve", "my_permissions"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": "create",
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_perms:ansible.add_ansiblerepository",
+            },
+            {
+                "action": "destroy",
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:ansible.delete_ansiblerepository",
+            },
+            {
+                "action": ["update", "partial_update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.change_ansiblerepository",
+                    "has_model_or_obj_perms:ansible.view_ansiblerepository",
+                ],
+            },
+            {
+                "action": ["modify"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.modify_ansible_repo_content",
+                    "has_model_or_obj_perms:ansible.view_ansiblerepository",
+                ],
+            },
+            {
+                "action": ["rebuild_metadata"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.rebuild_metadata_ansiblerepository",
+                ],
+            },
+            {
+                "action": ["repair"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.repair_ansiblerepository"
+                    "has_model_or_obj_perms:ansible.view_ansiblerepository",
+                ],
+            },
+            {
+                "action": ["sign"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.sign_ansiblerepository",
+                    "has_model_or_obj_perms:ansible.view_ansiblerepository",
+                ],
+            },
+            {
+                "action": ["sync"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
```

Comment:
Need to add three more conditions `has_remote_param_model_or_obj_perms:ansible.view_(collection/role/git)remote`. This makes sure the user has permission to use the remote to sync.

Reply:
@gerrod, when I add the three conditions my sync test fails (500, `RuntimeError: Permission ansible.view_roleremote is not suitable for objects of class <class 'pulp_ansible.app.models.CollectionRemote'>.`) due to all of the conditions needing to be met.  When only adding the condition for the type of remote being used (CollectionRemote in this case) it works as expected.

Reply:
Sounds like a bug in pulpcore, it doesn't account for repositories that can be synced from multiple remotes. I would just remove the check for now, we can add it back later once we fix it in pulpcore.

Reply:
Can't we add or statements in the conditions? Sure, we would need to prevent a 500 and get a 403 instead.

Reply:
Thanks @mdellweg, I tracked down `condition_expression` and implemented that to return 403 instead of the 500.